r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -493,3 +493,81 @@ test_that("edge cases in string detection and replacement", {
     tibble(x = c("ABC"))
   )
 })
+
+test_that("strptime", {
+
+  t_string <- tibble(x = c("2018-10-07 19:04:05", NA))
+  t_stamp <- tibble(x = c(lubridate::ymd_hms("2018-10-07 19:04:05"), NA))
+  t_stampPDT <- tibble(x = c(lubridate::ymd_hms("2018-10-07 19:04:05", tz = "PDT"), NA))
```

Comment:
I missed a typo in code, line 554: `t_stampPDT` should be used instead of `t_stamp`.

But it relates to your [comment 1](https://github.com/apache/arrow/pull/10334/files#r640822991) and [comment 2](https://github.com/apache/arrow/pull/10334#pullrequestreview-670405891).

I added example `t_stampPDT` to the test to see if I get a warning as `tz` agrument is given. I do but then data is pulled into `R`. Test now correctly fails as `lubridate` converts time to match PDT time zone. But then it should `stop()` as Neal [suggested](https://github.com/apache/arrow/pull/10334#discussion_r635544090) but I am not sure I know how to do that.

Adding separate test to check if we error correctly could be something in the lines of:

```
test_that("errors in strptime", {
  # Error when tz is passed

  x <- Expression$field_ref("x")
  expect_error(
    nse_funcs$strptime(x, tz = "PDT"),
    'Time zone argument not supported by Arrow'
  )
})
```

and then lines from [comment](https://github.com/apache/arrow/pull/10334/files#r640822991) are redundant.

Reply:
I think that that's what we want actually. 

In many other cases where something isn't (yet) supported in Arrow we automatically pull the data into R with a warning (in some circumstances like this). You might have found this already, but the pattern you propose for the test in your comment matches what we do elsewhere https://github.com/apache/arrow/blob/master/r/tests/testthat/test-dplyr-string-functions.R#L360-L369 which is good (comments about that test also have a bit more explanation about what's going on when the data warnings+is pulled in)